lib/spack/env/cc

```diff
@@ -712,10 +731,25 @@ esac
 # Linker flags
 case "$mode" in
     ld|ccld)
-        extend flags_list SPACK_LDFLAGS
+        extend spack_flags_list SPACK_LDFLAGS
         ;;
 esac
 
+IFS="$lsep"
+    categorize_arguments $spack_flags_list
+unset IFS
+        spack_flags_include_dirs_list="$return_include_dirs_list"
```

Comment:
This and the changes below look functional, but why double up all the extends on args_list rather than putting the user's flags at the end of each of these flags lists here? Ideally these would be reordered to find the spack flags first, add those, then extend with the users so we don't end up with all the extra branches at the end.